packages/openai-compatible/src/openai-compatible-chat-settings.ts

```diff
@@ -0,0 +1,20 @@
+export type OpenAICompatibleChatModelId = string;
+
+export interface OpenAICompatibleChatSettings {
+  /**
+A unique identifier representing your end-user, which can help the provider to
+monitor and detect abuse.
+  */
+  user?: string;
+
+  /**
+Default object generation mode that should be used with this model when
+no mode is specified. Should be the mode with the best results for this
+model. `undefined` can be returned if object generation is not supported.
+
+This is needed to generate the best objects possible w/o requiring the
+user to explicitly specify the object generation mode.
+  */
+  // TODO(shaper): This is really model-specific, move to config or elsewhere?
+  defaultObjectGenerationMode?: 'json' | 'tool' | undefined;
```

Comment:
can you make it a constructor parameter for the model that is then defined in the providers? (which would have that knowledge)

Reply:
By `options` do you mean `config` e.g. `OpenAICompatibleChatLanguageModel` constructor signature is `(modelId, settings, config)`?

I'll move it there and see how to expose it to the concrete provider impl which today just passes a settings object (where I had it when you commented), but we could add an options or model-config arg as well.